src/Engine/Renderer/RenderTechnique/RenderTechnique.hpp

```diff
@@ -19,6 +19,7 @@ namespace Ra
     namespace Engine
     {
         // NOTE(Charly): Would it be interesting to provide more stuff here ?
+        // TODO(Val) : we need proper memory management of these objects.

```

Comment:
I think that the raw pointer const ShaderProgram* shader must become shared_ptr<ShaderProgram> to have proper memory management. The class might remains but bettter memory management must be fixed.